snippets/defer.md

```diff
@@ -0,0 +1,21 @@
+### defer
+
+Defers invoking a function until the current call stack has cleared.
+
+Use `setTimeout()` with a timeout of 1ms to add a new event to the browser
+event queue and allow the rendering engine to complete its work. Use the spread/rest (`...`)
+operator to supply the function with an arbitrary number of arguments.
+
+```js
+const defer = (fn, ...args) => setTimeout(fn, 1, ...args);
```

Comment:
Is 1 ms required? Unless setTimeout has changed shouldn't a 0ms still cause it to wait until stack clear?
Also is there any reason it isn't curried?(Just personal taste here, non curried is also perfectly fine)
```js
const defer => fn => (...args) => setTimeout(fn, 0, ...args)
```

Reply:
```js
const defer = (fn, ...args) => setTimeout(fn, 0, ...args)
defer(console.log, 'a'), console.log('b')
-> b
<- undefined
-> a
```

Reply:
Lodash uses `1ms` for whatever reason, just copied it

Reply:
I wonder why. When I check the spec for the current setTimeout it had diverged from window.nextTick so 0 ms was now possible. I'm curious why lodash chose 1ms. Still this seems fine. I do like the idea of deferring it for later with a closure, but deferring it directly in line is fine too since a bind can be used for the same effect.